mpi-proxy-split/test/run_mana_mpi_regression_test.py

```diff
@@ -0,0 +1,210 @@
+import sys
+import os
+import subprocess
+import time
+import argparse
+from threading import Thread
+
+from subprocess import PIPE
+
+verbose = False
+
+# Wrapper to eprint to stderr
+def eprint(*args):
+    if verbose:
+        print(*args, file=sys.stderr)
+
+# Custom argparser to eprint error message
+class CustomParser(argparse.ArgumentParser):
+    def error(self, message):
+        sys.stderr.write('error: %s\n' % message)
+        self.print_help()
+        os._exit(2)
+
+# Error handling thread for timeouts
+def timeout_exit(timeout):
+    time.sleep(int(timeout))
+    sys.stderr.write('ERROR: TEST FAILED BY TIMEOUT\n')
+    sys.stderr.flush()
+    os._exit(11)
+
+# Remove unnecessary files (old checkpoint files)
+def clean():
+    subprocess.run('rm -rf ckpt_rank_*', shell=True)
+    subprocess.run('rm -rf dmtcp_restart_*', shell=True)
+
+# Helper function to get the number of currently running ranks
+def get_running_ranks(mana_dir, state=None):
+    status = subprocess.run([f"{mana_dir}/bin/mana_status"], stdout=PIPE,
+                             stderr=subprocess.DEVNULL)
+    if state is None:
+        return status.stdout.decode('utf-8').count(".mana.exe")
+    else:
+        return status.stdout.decode('utf-8').count(state)
+
+'''
+Example usage:
+
+python3 -n 3 -m ~/mana -n 4 ~/mana Allgather_test -t 10
```

Comment:
This usage is not working for me! There's no script name in it. The `-n` option and `~/mana` is twice.

Reply:
fixed